### PR TITLE
[AGENT-4722] Add Airflow DAG examples and Update documentation for new operators

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Added and update documentation for new operators
Added Airflow DAG examples with sample configurations

## Rationale
Users should be able to get documentation for new operators with DAG examples

### CODE REVIEW

**Review by elatt — APPROVED** [review]

**Review by alexromanyuk — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+"replicas": - A fixed number of replicas that will be deployed in the cluster.
+```
+
+For more [custom model test creation parameters](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/mlops/custom_model.html?highlight=CustomModelTest#create-custom-model-test), see the DataRobot documentation.
+
+---
+
+#### `GetCustomModelTestOverallStatusOperator`
+
+Get a custom model testing overall status.
```
Comment (nate-goudreault) [inline README.md:957]:
```suggestion
Get the overall status for custom model tests.
```

File: README.md [hunk 2]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Create a deployment from a DataRobot custom model image.
+
+Returns the created deployment id
+
+Parameters:
+
+| Parameter                      | Type | Description                                                                      |
+|--------------------------------|------|----------------------------------------------------------------------------------|
+| `custom_model_version_id`      | str  | The id of the DataRobot custom model version to deploy                           |
```
Comment (nate-goudreault) [inline README.md:981]:
```suggestion
| `custom_model_version_id`      | str  | The ID of the deployed custom model.                           |
```

File: README.md [hunk 3]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+| `deployment_name`              | str  | a human readable label (name) of the deployment                                  |
+| `default_prediction_server_id` | str  | an identifier of a prediction server to be used as the default prediction server |
+| `description`                  | str  | a human readable description of the deployment                                   |
+| `importance`                   | str  | deployment importance                                                            |
```
Comment (nate-goudreault) [inline README.md:985]:
```suggestion
| `default_prediction_server_id` | str  | An identifier for the default prediction server. |
| `description`                  | str  | A human-readable description of the deployment.                                   |
| `importance`                   | str  | The deployment importance level.                                                          |
```

File: README.md [hunk 4]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+|-----------------|------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `deployment_id` | str  | DataRobot deployment ID                                                                                                                                                                                                             |
+| `new_model_id`  | str  | The id of the new model to use. If replacing the deployment's model with a CustomInferenceModel, a specific CustomModelVersion ID must be used.                                                                                     |
+| `reason`        | str  | The reason for the model replacement. Must be one of 'ACCURACY', 'DATA_DRIFT', 'ERRORS', 'SCHEDULED_REFRESH', 'SCORING_SPEED', or 'OTHER'. This value will be stored in the model history to keep track of why a model was replaced |
```
Comment (nate-goudreault) [inline README.md:1019]:
```suggestion
| `reason`        | str  | The reason for the model replacement. Must be one of 'ACCURACY', 'DATA_DRIFT', 'ERRORS', 'SCHEDULED_REFRESH', 'SCORING_SPEED', or 'OTHER'. This value will be stored in the model history to keep track of why a model was replaced. |
```

File: README.md [hunk 5]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Activate or deactivate a Deployment.
+
+Returns the Deployment status (active/inactive)
+
+Parameters:
+
+| Parameter               | Type | Description                                                                   |
+|-------------------------|------|-------------------------------------------------------------------------------|
+| `deployment_id`         | str  | DataRobot deployment ID                                                       |
```
Comment (nate-goudreault) [inline README.md:1035]:
```suggestion
| `deployment_id`         | str  | An identifier for the deployed model.                                                      |
```

File: README.md [hunk 6]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+| `activate`              | str  | if set to True - activate deployment, if set to False - deactivate deployment |
```
Comment (nate-goudreault) [inline README.md:1036]:
```suggestion
| `activate`              | str  | If set to True, this parameter activates the deployment. Set to False to deactivate the deployment. |
```

File: README.md [hunk 7]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [activate deployment](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=replace_model#datarobot.models.Deployment.activate), see the DataRobot documentation.
```
Comment (nate-goudreault) [inline README.md:1038]:
```suggestion
For more [activate deployment parameters](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=replace_model#datarobot.models.Deployment.activate), see the DataRobot documentation.
```

File: README.md [hunk 8]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+---
+
+#### `GetDeploymentStatusOperator`
+
+Get a Deployment status (active/inactive).
+
+Returns the Deployment status (active/inactive)
```
Comment (nate-goudreault) [inline README.md:1046]:
```suggestion
Get the deployment status (active or inactive).

Returns the deployment status.
```

File: README.md [hunk 9]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter       | Type | Description              |
+|-----------------|------|--------------------------|
+| `deployment_id` | str  | DataRobot deployment ID  |
```
Comment (nate-goudreault) [inline README.md:1052]:
```suggestion
| `deployment_id` | str  | An identifier for the deployed model.  |
```

File: README.md [hunk 10]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [deployment](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=replace_model#datarobot.models.Deployment.activate), see the DataRobot documentation.
```
Comment (nate-goudreault) [inline README.md:1054]: deployment what? parameters? 
```suggestion
For more [deployment parameters](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=replace_model#datarobot.models.Deployment.activate), see the DataRobot documentation.
```

**Review by nate-goudreault — CHANGES_REQUESTED** [review]
Sharing first half of my comments now, and will resume with more edits.

File: README.md [hunk 11]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+| Parameter              | Type | Description                                                                                                      |
+|------------------------|------|------------------------------------------------------------------------------------------------------------------|
+| `dataset_identifier`   | str  | Alias of the dataset (used directly as part of the generated feature names)                                      |
+| `dataset_id`           | str  | Identifier of the dataset in DataRobot AI Catalog                                                                |
+| `dataset_version_id`   | str  | Identifier of the dataset version in DataRobot AI Catalog                                                        |
+| `primary_temporal_key` | str  | Name of the column indicating time of record creation                                                            |
+| `feature_list_id`      | str  | Specifies which feature list to use.                                                                             |
+| `snapshot_policy`      | str  | Policy to use  when creating a project or making predictions. If omitted, by default endpoint will use 'latest'. |
```
Comment (nate-goudreault) [inline README.md:1092]:
```suggestion
| `dataset_id`           | str  | The identifier of the dataset in the AI Catalog.                                                                |
| `dataset_version_id`   | str  | The identifier of the dataset version in the AI Catalog.                                                        |
| `primary_temporal_key` | str  | The name of the column indicating the time of record creation.                                                            |
| `feature_list_id`      | str  | Specifies the feature list to use.                                                                             |
| `snapshot_policy`      | str  | The policy to use when creating a project or making predictions. If omitted, the endpoint will use 'latest' by default. |
```

File: README.md [hunk 12]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [create-dataset-definitions-and-relationships-using-helper-functions](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/data/feature_discovery.html?highlight=DatasetDefinition#create-dataset-definitions-and-relationships-using-helper-functions), see the DataRobot documentation.
+
+---
+
+#### `DatasetRelationshipOperator`
+
+Create a Relationship between dataset defined in DatasetDefinition.
+
+Returns Dataset definition dict
```
Comment (nate-goudreault) [inline README.md:1103]:
```suggestion
Returns a dataset definition dict.
```

File: README.md [hunk 13]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+| `snapshot_policy`      | str  | Policy to use  when creating a project or making predictions. If omitted, by default endpoint will use 'latest'. |
+
+
+For more [create-dataset-definitions-and-relationships-using-helper-functions](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/data/feature_discovery.html?highlight=DatasetDefinition#create-dataset-definitions-and-relationships-using-helper-functions), see the DataRobot documentation.
+
+---
+
+#### `DatasetRelationshipOperator`
+
+Create a Relationship between dataset defined in DatasetDefinition.
```
Comment (nate-goudreault) [inline README.md:1101]:
```suggestion
Create a relationship between datasets defined in DatasetDefinition.
```

File: README.md [hunk 14]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+|---------------------------------------|------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `dataset1_identifier`                 | List[str]  | Identifier of the first dataset in this relationship. This is specified in the identifier field of dataset_definition structure. If None, then the relationship is with the primary dataset.                                                                                            |
+| `dataset2_identifier`                 | List[str]  | Identifier of the second dataset in this relationship. This is specified in the identifier field of dataset_definition schema.                                                                                                                                                          |
```
Comment (nate-goudreault) [inline README.md:1110]:
```suggestion
| `dataset2_identifier`                 | List[str]  | Identifier of the second dataset in this relationship. This is specified in the identifier field of the `dataset_definition` schema.                                                                                                                                                          |
```

File: README.md [hunk 15]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Create a Relationship between dataset defined in DatasetDefinition.
+
+Returns Dataset definition dict
+
+Parameters:
+
+| Parameter                             | Type       | Description                                                                                                                                                                                                                                                                             |
+|---------------------------------------|------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `dataset1_identifier`                 | List[str]  | Identifier of the first dataset in this relationship. This is specified in the identifier field of dataset_definition structure. If None, then the relationship is with the primary dataset.                                                                                            |
```
Comment (nate-goudreault) [inline README.md:1109]:
```suggestion
| `dataset1_identifier`                 | List[str]  | Identifier of the first dataset in this relationship. This is specified in the identifier field of the `dataset_definition` structure. If set to None, then the relationship is with the primary dataset.                                                                                            |
```

File: README.md [hunk 16]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+| `feature_derivation_windows`          | List       | List of feature derivation windows settings. If present, time-aware joins will be used. Only allowed when feature_derivation_window_start, feature_derivation_window_end and feature_derivation_window_time_unit are not provided.                                                      |
+| `prediction_point_rounding`           | List[dict] | Closest value of prediction_point_rounding_time_unit to round the prediction point into the past when applying the feature deri if present.Only applicable when dataset1_identifier is not provided.                                                                                    |
+| `prediction_point_rounding_time_unit` | str        | Time unit of the prediction point rounding. One of ``datarobot.enums.AllowedTimeUnitsSAFER`` Only applicable when dataset1_identifier is not provided.                                                                                                                                  |
```
Comment (nate-goudreault) [inline README.md:1118]:
```suggestion
| `dataset1_keys`                       | List[str]  | A list of strings (max length: 10 min length: 1). The column(s) from the first dataset which are used to join to the second dataset.                                                                                                                                                            |
| `dataset2_keys`                       | List[str]  | A list of strings (max length: 10 min length: 1). The column(s) from the second dataset that are used to join to the first dataset.                                                                                                                                                             |
| `feature_derivation_window_start`     | int        | How many time units of each dataset's primary temporal key into the past relative to the datetimePartitionColumn the feature derivation window should begin. Will be a negative integer, If present, the feature engineering graph performs time-aware joins.                       |
| `feature_derivation_window_end`       | int        | Determines how many units of time of each dataset's record primary temporal key into the past relative to the datetimePartitionColumn the feature derivation window should end.  It is a non-positive integer if present. If present, the feature engineering graph performs time-aware joins. |
| `feature_derivation_window_time_unit` | str        | The unit of time the feature derivation window. One of ``datarobot.enums.AllowedTimeUnitsSAFER`` If present, time-aware joins will be used. Only applicable when dataset1_identifier is not provided.                                                                                       |
| `feature_derivation_windows`          | List       | A list of feature derivation windows settings. If present, time-aware joins will be used. Only allowed when `feature_derivation_window_start`, `feature_derivation_window_end `, and `feature_derivation_window_time_unit` are not provided.                                                      |
| `prediction_point_rounding`           | List[dict] | Closest value of `prediction_point_rounding_time_unit` to round the prediction point into the past when applying the feature derivation if present. Only applicable when `dataset1_identifier` is not provided.                                                                                    |
| `prediction_point_rounding_time_unit` | str        | Time unit of the prediction point rounding. One of ``datarobot.enums.AllowedTimeUnitsSAFER``. Only applicable when `dataset1_identifier` is not provided.                                                                                                                                  |
```

File: README.md [hunk 17]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+
+For more [create-dataset-definitions-and-relationships-using-helper-functions](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/data/feature_discovery.html?highlight=DatasetDefinition#create-dataset-definitions-and-relationships-using-helper-functions), see the DataRobot documentation.
+
+---
+
+#### `ComputeFeatureImpactOperator`
+
+Creates Feature Impact job in DataRobot.
```
Comment (nate-goudreault) [inline README.md:1127]:
```suggestion
Creates a Feature Impact job in DataRobot.
```

File: README.md [hunk 18]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Returns Feature Impact job ID
```
Comment (nate-goudreault) [inline README.md:1129]:
```suggestion
Returns a Feature Impact job ID.
```

File: README.md [hunk 19]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter       | Type | Description           |
+|-----------------|------|-----------------------|
+| `project_id`    | str  | DataRobot project ID  |
+| `model_id`      | str  | DataRobot model ID    |
```
Comment (nate-goudreault) [inline README.md:1136]:
```suggestion
| `project_id`    | str  | DataRobot project ID.  |
| `model_id`      | str  | DataRobot model ID.    |
```

File: README.md [hunk 20]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [request_feature_impact](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=request_feature_impact#datarobot.models.Model.request_feature_impact), see the DataRobot documentation.
+
+---
+
+#### `ComputeFeatureEffectsOperator`
+
+Submit request to compute Feature Effects for the model.
```
Comment (nate-goudreault) [inline README.md:1144]:
```suggestion
Submit a request to compute Feature Effects for the model.
```

File: README.md [hunk 21]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Returns Feature Effects job ID
```
Comment (nate-goudreault) [inline README.md:1146]:
```suggestion
Returns the Feature Effects job ID.
```

File: README.md [hunk 22]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter        | Type | Description          |
+|------------------|------|----------------------|
+| `project_id`     | str  | DataRobot project ID |
+| `model_id`       | str  | DataRobot model ID   |
```
Comment (nate-goudreault) [inline README.md:1153]:
```suggestion
| `project_id`     | str  | DataRobot project ID. |
| `model_id`       | str  | DataRobot model ID.   |
```

File: README.md [hunk 23]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [request_feature_impact](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=request_feature_effect#datarobot.models.DatetimeModel.request_feature_effect), see the DataRobot documentation.
```
Comment (nate-goudreault) [inline README.md:1155]: incomplete, parameters or functions?

File: README.md [hunk 24]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+Submit request to compute SHAP impact job for the model.
+
+Returns SHAP impact job ID
+
+Parameters:
+
+| Parameter      | Type | Description             |
+|----------------|------|-------------------------|
+| `project_id`   | str  | DataRobot project ID    |
+| `model_id`     | str  | DataRobot model ID      |
```
Comment (nate-goudreault) [inline README.md:1170]:
```suggestion
| `project_id`   | str  | DataRobot project ID.    |
| `model_id`     | str  | DataRobot model ID.      |
```

File: README.md [hunk 25]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [shap-impact](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/modeling/model.html?highlight=ShapImpact#shap-impact), see the DataRobot documentation.
```
Comment (nate-goudreault) [inline README.md:1172]: incomplete

File: README.md [hunk 26]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+---
+
+#### `CreateExternalModelPackageOperator`
+
+Create an external model package in DataRobot MLOps from JSON configuration.
+
+Returns a model package ID of newly created ModelPackage.
```
Comment (nate-goudreault) [inline README.md:1180]:
```suggestion
Returns a model package ID of the newly created model package.
```

File: README.md [hunk 27]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+        }
+    }
+
+---
+
+#### `DeployModelPackageOperator`
+
+Create a deployment from a DataRobot model package.
+
+Returns The created deployment ID
```
Comment (nate-goudreault) [inline README.md:1250]:
```suggestion
Returns the created deployment ID.
```

File: README.md [hunk 28]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter                      | Type           | Description                                                                                                                                                                      |
+|--------------------------------|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `deployment_name`              | str            | A human readable label of the deployment.                                                                                                                                        |
+| `model_package_id`             | str            | The ID of the DataRobot model package to deploy.                                                                                                                                 |
+| `default_prediction_server_id` | str            | An identifier of a prediction server to be used as the default prediction server. When working with prediction environments, default prediction server Id should not be provided |
```
Comment (nate-goudreault) [inline README.md:1258]:
```suggestion
| `default_prediction_server_id` | str            | An identifier of a prediction server to be used as the default prediction server. When working with prediction environments, the default prediction server ID should not be provided. |
```

File: README.md [hunk 29]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+| `description`                  | str            | A human readable description of the deployment.                                                                                                                                  |
+| `importance`                   | str            | Deployment importance level.                                                                                                                                                     |
+| `user_provided_id`             | str            | A user-provided unique ID associated with a deployment definition in a remote git repository.                                                                                    |
+| `additional_metadata`          | Dict[str, str] | A Key/Value pair dict, with additional metadata.                                                                                                                                 |
+
+---
+
+#### `AddExternalDatasetOperator`
+
+Upload a new dataset from a catalog dataset to make predictions for a model
```
Comment (nate-goudreault) [inline README.md:1269]:
```suggestion
Upload a new dataset from the AI Catalog to make predictions for a model.
```

File: README.md [hunk 30]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Returns external dataset ID for the model
```
Comment (nate-goudreault) [inline README.md:1271]:
```suggestion
Returns an external dataset ID for the model,
```

File: README.md [hunk 31]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter            | Type | Description                              |
+|----------------------|------|------------------------------------------|
+| `project_id`         | str  | DataRobot project ID                     |
+| `dataset_id`         | str  | DataRobot AI Catalog dataset ID          |
+| `credential_id`      | str  | DataRobot Credentials ID                 |
+| `dataset_version_id` | str  | DataRobot AI Catalog dataset version ID  |
```
Comment (nate-goudreault) [inline README.md:1280]:
```suggestion
| `project_id`         | str  | DataRobot project ID.                     |
| `dataset_id`         | str  | DataRobot AI Catalog dataset ID.          |
| `credential_id`      | str  | DataRobot credentials ID.                 |
| `dataset_version_id` | str  | DataRobot AI Catalog dataset version ID.  |
```

File: README.md [hunk 32]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [upload_dataset_from_catalog](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=upload_dataset_from_catalog#datarobot.models.Project.upload_dataset_from_catalog), see the DataRobot documentation.
```
Comment (nate-goudreault) [inline README.md:1282]: functions or parameters?

File: README.md [hunk 33]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+---
+
+#### `RequestModelPredictionsOperator`
+
+Requests predictions against a previously uploaded dataset.
+
+Returns model predictions job ID
```
Comment (nate-goudreault) [inline README.md:1290]:
```suggestion
Returns a model predictions job ID.
```

File: README.md [hunk 34]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter             | Type | Description                   |
+|-----------------------|------|-------------------------------|
+| `project_id`          | str  | DataRobot project ID          |
+| `model_id`            | str  | DataRobot model ID            |
+| `external_dataset_id` | str  | DataRobot external dataset ID |
```
Comment (nate-goudreault) [inline README.md:1298]:
```suggestion
| `project_id`          | str  | DataRobot project ID.          |
| `model_id`            | str  | DataRobot model ID.            |
| `external_dataset_id` | str  | DataRobot external dataset ID. |
```

File: README.md [hunk 35]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+For more [request_predictions](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/autodoc/api_reference.html?highlight=request_predictions#datarobot.models.Model.request_predictions), see the DataRobot documentation.
+
+---
+
+#### `TrainModelOperator`
+
+Submit a job to the queue to train a model from specific blueprint.
```
Comment (nate-goudreault) [inline README.md:1306]:
```suggestion
Submit a job to the queue to train a model from a specific blueprint.
```

File: README.md [hunk 36]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Returns model training job ID
```
Comment (nate-goudreault) [inline README.md:1308]:
```suggestion
Returns a model training job ID.
```

File: README.md [hunk 37]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter              | Type | Description                                                                                                                                             |
+|------------------------|------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `project_id`           | str  | DataRobot project ID                                                                                                                                    |
+| `blueprint_id`         | str  | DataRobot blueprint ID                                                                                                                                  |
+| `featurelist_id`       | str  | The identifier of the featurelist to use. If not defined, the default for this project is used.                                                         |
+| `source_project_id`    | str  | Which project created this blueprint_id. If ``None``, it defaults to looking in this project. Note that you must have read permissions in this project. |
```
Comment (nate-goudreault) [inline README.md:1317]:
```suggestion
| `project_id`           | str  | DataRobot project ID.                                                                                                                                    |
| `blueprint_id`         | str  | DataRobot blueprint ID.                                                                                                                                  |
| `featurelist_id`       | str  | The identifier of the feature list to use. If not defined, the default feature list for this project is used.                                                         |
| `source_project_id`    | str  | The source project that created the `blueprint_id`. If ``None``, it defaults to looking in this project. Note that you must have read permissions in this project. |
```

File: README.md [hunk 38]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+    "n_clusters":
+}
+
+For more [start-training-a-model](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/modeling/model.html?highlight=project.train#start-training-a-model), see the DataRobot documentation.
+
+---
+
+#### `RetrainModelOperator`
+
+Submit a job to the queue to retrain a model on a specific sample size and/or custom featurelist.
```
Comment (nate-goudreault) [inline README.md:1333]:
```suggestion
Submit a job to the queue to retrain a model on a specific sample size and/or custom feature list.
```

File: README.md [hunk 39]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Returns a model retraining job ID
```
Comment (nate-goudreault) [inline README.md:1335]:
```suggestion
Returns a model retraining job ID.
```

File: README.md [hunk 40]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Parameters:
+
+| Parameter            | Type | Description                                                                                     |
+|----------------------|------|-------------------------------------------------------------------------------------------------|
+| `project_id`         | str  | DataRobot project ID                                                                            |
+| `model_id`           | str  | DataRobot model ID                                                                              |
+| `featurelist_id`     | str  | The identifier of the featurelist to use. If not defined, the default for this project is used. |
```
Comment (nate-goudreault) [inline README.md:1343]:
```suggestion
| `project_id`         | str  | DataRobot project ID.                                                                            |
| `model_id`           | str  | DataRobot model ID.                                                                              |
| `featurelist_id`     | str  | The identifier of the feature list to use. If not defined, the default for this project is used. |
```

File: README.md [hunk 41]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+    "training_row_count":
+}
+
+For more [train-a-model-on-a-different-sample-size](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/modeling/model.html?highlight=model.train#train-a-model-on-a-different-sample-size), see the DataRobot documentation.
+
+---
+
+#### `PredictionExplanationsInitializationOperator`
+
+Triggering a prediction explanations initialization of a model.
```
Comment (nate-goudreault) [inline README.md:1358]:
```suggestion
Initialize prediction explanations for a model.
```

File: README.md [hunk 42]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+Returns a Prediction Explanations Initialization job ID
+
+Parameters:
+
+| Parameter            | Type | Description          |
+|----------------------|------|----------------------|
+| `project_id`         | str  | DataRobot project ID |
+| `model_id`           | str  | DataRobot model ID   |
```
Comment (nate-goudreault) [inline README.md:1367]:
```suggestion
| `project_id`         | str  | DataRobot project ID. |
| `model_id`           | str  | DataRobot model ID.   |
```

File: README.md [hunk 43]
```diff
@@ -832,6 +1444,34 @@ Parameters:
 |-----------------------------|------|-------------|
 | `job_id`                    | str  | The DataRobot async API call status check ID. |
 
+---
+
+#### `DataRobotJobSensor`
+
+Checks whether DataRobot Job is complete.
```
Comment (nate-goudreault) [inline README.md:1451]:
```suggestion
Checks whether a DataRobot job is complete.
```

File: README.md [hunk 44]
```diff
@@ -832,6 +1444,34 @@ Parameters:
...
+
+Parameters:
+
+| Parameter                   | Type | Description           |
+|-----------------------------|------|-----------------------|
+| `project_id`                | str  | DataRobot project ID. |
+| `job_id`                    | str  | DataRobot Job ID.     |
```
Comment (nate-goudreault) [inline README.md:1458]:
```suggestion
| `job_id`                    | str  | DataRobot Job ID.     |
```
```suggestion
| `job_id`                    | str  | DataRobot job ID.     |
```

File: README.md [hunk 45]
```diff
@@ -832,6 +1444,34 @@ Parameters:
...
+Checks whether DataRobot Model Training Job is complete.
+
+Returns False if job not yet completed, PokeReturnValue(True, trained_model.id) if model training completed
+
+Parameters:
+
+| Parameter                   | Type | Description           |
+|-----------------------------|------|-----------------------|
+| `project_id`                | str  | DataRobot project ID. |
+| `job_id`                    | str  | DataRobot Job ID.     |
```
Comment (nate-goudreault) [inline README.md:1473]:
```suggestion
| `job_id`                    | str  | DataRobot job ID.     |
```

File: README.md [hunk 46]
```diff
@@ -854,31 +1494,46 @@ See the the [`datarobot_provider/example_dags`](https://github.com/datarobot/air
...
-| `datarobot_upload_actuals_catalog_dag.py`              | Upload actuals from the DataRobot AI Catalog. |
-| `deployment_service_stats_dag.py`                      | Get a deployment's service statistics with `GetServiceStatsOperator` |
-| `deployment_stat_and_accuracy_dag.py`                  | Get a deployment's service statistics and accuracy. |
-| `deployment_update_monitoring_settings_dag.py`         | Update a deployment's monitoring settings. |
-| `deployment_update_segment_analysis_settings_dag.py`   | Update a deployment's segment analysis settings. |
-| `download_scoring_code_from_deployment_dag.py`         | Download scoring code (JAR file) from a DataRobot deployment. |
-| `advanced_datarobot_pipeline_jdbc_dag.py`              | Run the advanced end-to-end workflow in DataRobot. |
+| `datarobot_dataset_new_version_dag.py`                 | Create a new version of an existing dataset in the DataRobot AI Catalog.                |
+| `datarobot_dataset_upload_dag.py`                      | Upload a local file to the DataRobot AI Catalog.                                        |
```
Comment (nate-goudreault) [inline README.md:1511]:
```suggestion
| `datarobot_dataset_upload_dag.py`                      | Upload a local file to the AI Catalog.                                        |
```

File: README.md [hunk 47]
```diff
@@ -854,31 +1494,46 @@ See the the [`datarobot_provider/example_dags`](https://github.com/datarobot/air
...
+| `datarobot_get_datastore_dag.py`                       | Create a DataRobot DataStore with `GetOrCreateDataStoreOperator`.                       |
```
Comment (nate-goudreault) [inline README.md:1512]:
```suggestion
| `datarobot_get_datastore_dag.py`                       | Create a DataRobot data store with `GetOrCreateDataStoreOperator`.                       |
```

File: README.md [hunk 48]
```diff
@@ -854,31 +1494,46 @@ See the the [`datarobot_provider/example_dags`](https://github.com/datarobot/air
...
+| `datarobot_jdbc_dataset_dag.py`                        | Create a DataRobot project from a JDBC data source.                                     |
+| `datarobot_jdbc_dynamic_dataset_dag.py`                | Create a DataRobot project from a JDBC dynamic data source.                             |
+| `datarobot_upload_actuals_catalog_dag.py`              | Upload actuals from the DataRobot AI Catalog.                                           |
+| `deployment_service_stats_dag.py`                      | Get a deployment's service statistics with `GetServiceStatsOperator`                    |
```
Comment (nate-goudreault) [inline README.md:1516]:
```suggestion
| `deployment_service_stats_dag.py`                      | Get a deployment's service statistics with `GetServiceStatsOperator`.                    |
```

**Review by nate-goudreault — CHANGES_REQUESTED** [review]

File: README.md [hunk 49]
```diff
@@ -854,31 +1494,46 @@ See the the [`datarobot_provider/example_dags`](https://github.com/datarobot/air
...
+| `datarobot_feature_discovery_pipeline_dag.py`          | Creating feature-discovery DataRobot project, train models pipline.                     |
+| `datarobot_timeseries_pipeline_dag.py`                 | Creating timeseries DataRobot project, train models pipline.                            |
+| `deployment_activate_deactivate_dag.py`                | Example of Deployment activation/deactivaion and get Deployment status.                 |
+| `deployment_replace_model_dag.py`                      | Example of Deployment model replacement.                                                |
+| `model_compute_insights_dag.py`                        | Example of compute FeatureImpact and FeatureEffects job.                                |
+| `model_compute_prediction_explanations_dag.py`         | Example of compute prediction explanations job.                                         |
+| `model_compute_predictions_dag.py`                     | Example of compute predictions for model.                                               |
+| `model_compute_shap_dag.py`                            | Example of compute SHAP job.                                                            |
```
Comment (nate-goudreault) [inline README.md:1534]:
```suggestion
| `datarobot_custom_model_pipeline_dag.py`               | Create an end-to-end workflow with custom models in DataRobot.                           |
| `datarobot_custom_partitioning_pipeline_dag.py`        | Create a custom partitioned project and train models.                              |
| `datarobot_datetime_partitioning_pipeline_dag.py`      | Create a datetime partitioned project.                            |
| `datarobot_external_model_pipeline_dag.py`             | An end-to-end workflow with external models in DataRobot.                         |
| `datarobot_feature_discovery_pipeline_dag.py`          | Create a Feature Discovery project and train models.                     |
| `datarobot_timeseries_pipeline_dag.py`                 | Create a time series DataRobot project.                            |
| `deployment_activate_deactivate_dag.py`                | An example of deployment activation/deactivation and getting deployment status.                 |
| `deployment_replace_model_dag.py`                      | An example of model replacement for deployments.                                                |
| `model_compute_insights_dag.py`                        | An example of computing Feature Impact and Feature Effects.                                |
| `model_compute_prediction_explanations_dag.py`         | An example of a compute prediction explanations job.                                         |
| `model_compute_predictions_dag.py`                     | An example of computing predictions for model.                                               |
| `model_compute_shap_dag.py`                            | An example of computing SHAP.                                                            |
```

File: README.md (hunk 25 above)
Comment (oleksandr-saienko) [reply README.md:1172]: what about "For more information" or "For more details"?

File: README.md (hunk 23 above)
Comment (oleksandr-saienko) [reply README.md:1155]: what about "For more information" or "For more details"?

File: README.md [hunk 50]
```diff
@@ -784,6 +784,618 @@ Sample config params:
...
+
+---
+
+#### `CreateCustomModelVersionOperator`
+
+Create a custom model version.
+
+Returns a created custom model version ID
+
+Parameters:
```
[the remark this answers is not in the document]
Comment (oleksandr-saienko) [reply README.md:899]: Do you mean to apply  this for all "Parameters" header?

**Review by nate-goudreault — APPROVED** [review]

File: README.md (hunk 50 above)
[the remark this answers is not in the document]
Comment (nate-goudreault) [reply README.md:899]: Yes but I didn't realize how many sections there were, no need for this change.

**Pull request merged**